internal/controller/eventing/eventmesh.go

```diff
@@ -59,13 +57,31 @@ func (r *Reconciler) reconcileEventMeshSubManager(ctx context.Context, eventing
 		return fmt.Errorf("failed to setup environment variables for EventMesh controller: %v", err)
 	}
 
+	// Read the cluster domain from the Eventing CR, or
+	// read it from the configmap managed by gardener
+	domain := eventing.Spec.Backend.Config.Domain
+	if utils.IsEmpty(domain) {
+		r.namedLogger().Infof(
+			`Domain is not configured in the Eventing CR, reading it from the ConfigMap %s/%s`,
+			shootInfoConfigMapNamespace, shootInfoConfigMapName,
+		)
+		domain, err = r.readDomainFromConfigMap(ctx)
+		if err != nil || utils.IsEmpty(domain) {
+			return domainMissingError(err)
+		}
+	}
+	r.namedLogger().Infof(`Domain is %s`, domain)
```

Comment:
Mybe we should also log fact that we got our domain from `eventing.Spec.Backend.Config.Domain` explicitly. While this is implicit, without seeing this code we would just see in in the logs what our domain is, without knowing, that it comes from `eventing.Spec.Backend.Config.Domain`.

Reply:
IMO these logs are not really needed, I just put them there for the first releases to help us see what the domain value is, then we should reduce them to debug level.

But for the time being, we will get one of the two following logs:

When the domain is configured in the Eventing CR, we get:

```
Domain is ...
```

Or, when it is not configured in the Eventing CR but it is configured in the ConfigMap, we get:

```
Domain is not configured in the Eventing CR ...
Domain is ...
```

What is more important is the condition we get from the Eventing CR:

```
  - lastTransitionTime: "2023-10-17T15:05:06Z"
    message: domain configuration is missing. domain must be configured in either the Eventing 
    CustomResource under "Spec.Backend.Config.Domain" or in the ConfigMap 
    "kube-system/shoot-info" under "data.domain"
    reason: EventMeshSubscriptionManagerFailed
    status: "False"
    type: SubscriptionManagerReady
```

which tells the user where to put the domain value.

Reply:
thanks for the detailed explanation. This is fine now :)